AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -361,8 +360,19 @@ - (void)_insertNodes:(NSArray *)nodes atIndexPaths:(NSArray *)indexPaths withAni
   [self insertNodes:nodes ofKind:ASDataControllerRowNodeKind atIndexPaths:indexPaths completion:^(NSArray *nodes, NSArray *indexPaths) {
     ASDisplayNodeAssertMainThread();
     
+    // TODO - Filter out Move indexPaths
+    NSMutableArray *newNodes = [[NSMutableArray alloc] initWithCapacity:nodes.count];
+    NSMutableArray *newIndexPaths = [[NSMutableArray alloc] initWithCapacity:indexPaths.count];
+    
+    for (int i = 0; i < indexPaths.count; i++) {
+      if (![_moveToIndexPaths containsObject:indexPaths[i]]) {
+        [newNodes addObject:nodes[i]];
+        [newIndexPaths addObject:indexPaths[i]];
+      }
+    }
+    
     if (_delegateDidInsertNodes)
```

Comment:
Very ugly here. It is trying to filter out the Move `from` (delete) and `to` (insert), and send them to UICollectionView move animation call.

Advices please.